docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md

```diff
@@ -59,6 +59,8 @@
     to
 
     `func NewClusterClient(c client.Client, cluster *clusterv1.Cluster, scheme runtime.Scheme) (client.Client, error)`
+- The returned client is no longer configured for lazy discovery. Any consumers that attempt to create 
```

Comment:
Doesn't the controller-runtime client do this by default now?

Reply:
controller-runtime configures the dynamic restmapper by default now, but it does not configure it with lazy discovery.

Reply:
We should probably keep this backward compatible, especially for clusterctl. In other situations as well we need the lazy discovery in Cluster API because the providers' CRDs could be installed after the client has been initialized

Reply:
The dynamic restmapper already handles that, the only thing that lazy discovery really gives you is the ability to create a client before the server is available for querying the discovery endpoint.

Reply:
I'd either make the function signature configurable, or keep the behavior the same for v1alpha3. wdyt?

Reply:
Do we know of an actual use case where we need lazy discovery rather than just the dynamic restmapper? As far as I know the old clusterctl workflow was maybe the only place we would have really needed it, and the new clusterctl workflow would not need the long lived client in the same way.

Reply:
I'm not sure if we're relying on this feature today, that said we're very close to v0.3.0 release and it sounds like we might need more time to vet this change making sure it doesn't impact different components.

How about if we make the change less invasive? We mark the behavior of setting the lazy client as deprecated, and we offer a configurable approach when creating the client to enable/disable it. 

This way, users of this API can opt in/out, but we guarantee today's behaviors.